File: src/main/java/com/listaai/infrastructure/adapter/output/persistence/repository/ListJpaRepository.java
package com.listaai.infrastructure.adapter.output.persistence.repository;

import com.listaai.infrastructure.adapter.output.persistence.entity.ListEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import java.util.List;

public interface ListJpaRepository extends JpaRepository<ListEntity, Long> {

    @Query("SELECT l FROM ListEntity l WHERE EXISTS " +
           "(SELECT us FROM UserShoppingListEntity us " +
           " WHERE us.id.listId = l.id AND us.id.userId = :userId)")
    List<ListEntity> findAllByUserId(@Param("userId") Long userId);
}

File: src/main/kotlin/com/listaai/infrastructure/adapter/output/persistence/repository/ListJpaRepository.kt
package com.listaai.infrastructure.adapter.output.persistence.repository

import com.listaai.infrastructure.adapter.output.persistence.entity.ListEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ListJpaRepository : JpaRepository<ListEntity, Long>